modules/core/04-channel/types/upgrade_test.go

```diff
@@ -0,0 +1,27 @@
+package types_test
+
+import (
+	errorsmod "cosmossdk.io/errors"
+
+	channeltypes "github.com/cosmos/ibc-go/v7/modules/core/04-channel/types"
+)
+
+func (suite *TypesTestSuite) TestUpgradeErrorIsOf() {
+	ue := channeltypes.NewUpgradeError(1, channeltypes.ErrInvalidChannel)
+	suite.Require().True(errorsmod.IsOf(ue, channeltypes.ErrInvalidChannel))
+	suite.Require().False(errorsmod.IsOf(ue, channeltypes.UpgradeError{}))
```

Comment:
We could add test cases also for:
- two empty upgrade errors with IsOf
- the err is empty upgrade error and target is non-empty

But I'm happy as is as well

Reply:
added! And refactored to use table tests